handbook/engineering/distribution/observability/monitoring_pillars.md

```diff
@@ -0,0 +1,196 @@
+
+Our monitoring infrastructure _forces_ you to declare at least a _Warning_ alert. It's OK if this is flaky, or even if it fires when something is not wrong.
+
+The point is _you have to give a best-guess at what a bad or good value looks like because you are the only one who can answer that question._ Otherwise, the information you are 'monitoring' cannot be interpreted by anyone else and is inherently useless to anyone except you: it should not exist in a dashboard then.
+
+#### What you should do instead
+
+Declare a `Warning` alert, and give it your best guess at what a threshold would look like. It's OK if it's flaky, or fires when there isn't a problem sometimes or on some Sourcegraph instances.
+
+Over time, we will adjust the alert so it becomes more reliable and becomes the baseline number that most Sourcegraph instances observe. Once it becomes really reliable (and empirically proven to be so, based on pings we get from customer instances), it can be promoted to a `Critical` alert as well if it speaks of a critical part of Sourcegraph.
+
+### FAQ: Why can't I create a dashboard in the WYSIWYG Grafana editor?
+
+This violates the third pillar of monitoring at Sourcegraph:
+
+> Creating dashboards outside of the monitoring generator, such as with the Grafana WYSIWYG editor, is forbidden. All metrics should be presented in the most plain, mundane, non-fanciful way that every site admin can understand.
+
+#### Why this is bad
+
+Using the Grafana WYSIWYG editor, or otherwise operating outside the constraints of [the monitoring generator](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/tree/monitoring) makes it:
+
+- Extremely easy to introduce dashboards that do not comply with the constraints we want to impose.
+- Extremely easy to introduce dashboards that have inconsistency errors through mistakes that are easy to make (e.g. not using UTC time on dashboards, the preset timeframe being inconsistent, it not self-describing to admins what it does, etc.)
+- Difficult for others to modify and review without introducing regressions (e.g. due to Grafana schema version changes, which happen frequently, producing large diffs)
+- Much harder to declare proper alerts in our regular high-level alerting architecture.
+
+#### What you should do instead
+
+You might be thinking:
+
+> But writing code for dashboards is quite painful, let alone a bunch of generation code that I now need to understand!
+
+In truth, we are only using Go as a nice type-checked declarative syntax. It's as simple to use as a configuration file, and the constraints we impose in monitoring means that we do not allow doing any fanciful dashboards intentionally -- which means you are basically _just_ writing a Prometheus query, and barely any layout or styling.
+
+See: [How easy is it to add monitoring?](monitoring.md#how-easy-is-it-to-add-monitoring)
+
+### FAQ: Why can't I create a graph/panel with more than 5 cardinality (labels)?
```

Comment:
Hmm.... I feel like these conflicts with other observability guidelines https://thenewstack.io/observability-a-3-year-retrospective/
From the article: 
> For example, observability is impossible without:

    raw events
    high cardinality dimensions
    no pre-aggregation, no pre-indexing (which lock you into asking predefined questions)
    read time aggregation

Reply:
On a second read-through, I think this is fine for _monitoring_. 
A section on the difference between monitoring and observability would be welcome though. 
These are terms that are still evolving though

Reply:
I do believe your quote is true!

The difference is that:

1. We get our raw events and cardinality from other tools that are better at handling that cardinality (e.g. logging, distributed tracing)
2. Monitoring is just one part of our observability stack, and in cases where higher cardinality in monitoring makes sense we _do allow it_ just not in individual graphs/panels (you need to split it into multiple panels/graphs for improved visibility).

Reply:
> A section on the difference between monitoring and observability would be welcome though.

Yes, definitely, I have an issue open here for me to add this: https://github.com/sourcegraph/sourcegraph/issues/9773

Reply:
Large part done in https://github.com/sourcegraph/about/pull/948